zerver/lib/actions.py

```diff
@@ -4534,6 +4570,30 @@ def do_update_message(user_profile: UserProfile, message: Message,
         )
         changed_messages += messages_list
 
+        if new_stream is not None:
+            assert stream_being_edited is not None
+
+            message_ids = [msg.id for msg in changed_messages]
+            # Delete UserMessage objects from guest users who will no
+            # longer have access to these messages.  Note: This could be
+            # very expensive, since it's N guest users x M messages.
+            UserMessage.objects.filter(
+                user_profile_id__in=[sub.user_profile_id for sub in
+                                     guest_subs_losing_access],
+                message_id__in=message_ids,
+            ).delete()
+
+            delete_event: DeleteMessagesEvent = {
+                'type': 'delete_message',
+                'message_ids': message_ids,
+                'message_type': 'stream',
+                'stream_id': stream_being_edited.id,
+                'topic': orig_topic_name,
+            }
+            delete_event_notify_user_ids = [sub.user_profile_id for sub in guest_subs_losing_access]
+            delete_event_notify_users = list(map(subscriber_info, delete_event_notify_user_ids))
+            send_event(user_profile.realm, delete_event, delete_event_notify_users)
```

Comment:
I think we should be passing `delete_event_notify_user_ids`; the `subscriber_info` map is a legacy interface.  From `event_queue.py`:

```
    elif event['type'] == "delete_message":                                                
        if len(users) > 0 and isinstance(users[0], dict):                                  
            # do_delete_messages used to send events with users in                         
            # dict format {"id": <int>} This block is here for                             
            # compatibility with events in that format still in the                        
            # queue at the time of upgrade.                                                
            #                                                                              
            # TODO: Remove this block in release >= 2.3.                                   
            user_ids: List[int] = [user['id'] for user in                                  
                                   cast(List[Mapping[str, int]], users)]                   
        else:                                                                              
            user_ids = cast(List[int], users)                                              
```

Reply:
Okay.